cellmlmanip/model.py

```diff
@@ -936,15 +936,16 @@ def connect_variables(self, source_name, target_name):
             target.assigned_to = source.assigned_to
             # Everywhere the target variable is used, replace with source variable,
             # updating the definition maps accordingly
-            for index, equation in enumerate(self.equations):
-                self.equations[index] = new_eq = equation.xreplace({target: source.assigned_to})
-                if equation.lhs.is_Derivative:
-                    state_var = equation.lhs.free_symbols.pop()
-                    assert state_var in self._ode_definition_map
-                    self._ode_definition_map[state_var] = new_eq
-                else:
-                    assert equation.lhs in self._var_definition_map
-                    self._var_definition_map[equation.lhs] = new_eq
+            if len(self.equations) > 0:
```

Comment:
my timing tests showed that it's faster to check len(dict) >0 than to enumerate over an empty dict.
Since 99% of the use is from the parser, I introduced this as a speed-up, I could have removed it entirely, but then we get a library method that doesn't quite do what you would expect, so that's why I kept it.

Reply:
How much does this save though?

Reply:
On my laptop time difference: 3.0e-07s vs  9.3e-08s in the example model we're calling this 81 times so it would save 
1.7e-05s so not really noticable to be fair. Come to think of it it doesn't seem to be used anywhere but the parser so maybe it's better to remove it altogether.